### PR TITLE
[AMORO_3053]Add system metric for ams service

### PR DESCRIPTION
**Pull request by xieyi888**
## Why are the changes needed?

Close https://github.com/apache/amoro/issues/3053.

## Brief change log

1. collect system metric for jvm  using api from jdk
it's refer from flink jvm system metric https://github.com/apache/flink/blob/af7d2b3ab0e0ecc4157005704bebac9c767f2e1a/flink-runtime/src/main/java/org/apache/flink/runtime/metrics/util/MetricUtils.java#L129
2. register metric in AmoroServiceContainer  during start ams httpserver by using amoro metric plugin  

## How was this patch tested?

- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

I had test HEAP_USED,HEAP_COMMITTED,HEAP_MAX,THREADS_COUNT ,the count collected by jdk api is meet expectations
But I'm not sure  the number of CPU_LOAD and CPU_TIME collected by jdk api is equals to the number collect from machinery system

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by czy006 — APPROVED** [review]
LGTM, These system metics help us better understand the current situation of AMS

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmsServiceMetrics.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+          (org.apache.amoro.api.metrics.Gauge<Long>) () -> garbageCollector.getCollectionTime());
+    }
+  }
+
+  private MetricDefine formatGarbageCollectorMetricDefine(
+      String garbageCollector, String identify) {
+    return defineGauge(
+            AMS_STATUS_JVM_GARBAGE_COLLECTOR
+                + "_"
+                + garbageCollector.replaceAll("\\s", "_")
```
Comment (baiyangtx) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmsServiceMetrics.java:149]: Maybe we should define grabage collector name as a tag?

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmsServiceMetrics.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+          .withDescription(
+              "The maximum amount of memory in the heap (in bytes), It's equal to the value specified through -Xmx")
+          .build();
+
+  public static final MetricDefine AMS_STATUS_JVM_THREADS_COUNT =
+      defineGauge("ams_status_jvm_threads_count")
+          .withDescription("The total number of live threads used by the AMS")
+          .build();
+
+  public static final String AMS_STATUS_JVM_GARBAGE_COLLECTOR = "ams_status_jvm_GarbageCollector";
```
Comment (baiyangtx) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmsServiceMetrics.java:67]: I think we should keep metric name style consistent.

ams_status_jvm_garbage_collector

**Review by baiyangtx — COMMENTED** [review]
I prefer metric prefix from `ams_status_jvm_` to `ams_jvm_`

**Comment by czy006** [comment]
Another question I have here is whether these metrics should also be improved when we use different GC recyclers, such as G1

**Comment by xieyi888** [comment]
> Another question I have here is whether these metrics should also be improved when we use different GC recyclers, such as G1



> Another question I have here is whether these metrics should also be improved when we use different GC recyclers, such as G1


```
    for (final GarbageCollectorMXBean garbageCollector : garbageCollectorMXBeans) {
      MetricDefine count = formatGarbageCollectorMetricDefine(garbageCollector.getName(), "count");
      MetricDefine time = formatGarbageCollectorMetricDefine(garbageCollector.getName(), "time");
      registerMetric(
          registry,
          count,
          (org.apache.amoro.api.metrics.Gauge<Long>) () -> garbageCollector.getCollectionCount());
      registerMetric(
          registry,
          time,
          (org.apache.amoro.api.metrics.Gauge<Long>) () -> garbageCollector.getCollectionTime());
    }
  }
```


> Another question I have here is whether these metrics should also be improved when we use different GC recyclers, such as G1

from  List<GarbageCollectorMXBean> will get  all the GarbageCollector 
and garbageCollector.getName() can get the name of GarbageCollector
if use G1 Collect, the garbageCollectorName will be  G1 Young and G1 Old
```
 List<GarbageCollectorMXBean> garbageCollectorMXBeans =
        ManagementFactory.getGarbageCollectorMXBeans();

    for (final GarbageCollectorMXBean garbageCollector : garbageCollectorMXBeans) {
      MetricDefine count = formatGarbageCollectorMetricDefine(garbageCollector.getName(), "count");
      MetricDefine time = formatGarbageCollectorMetricDefine(garbageCollector.getName(), "time");
      registerMetric(
          registry,
          count,
          (org.apache.amoro.api.metrics.Gauge<Long>) () -> garbageCollector.getCollectionCount());
      registerMetric(
          registry,
          time,
          (org.apache.amoro.api.metrics.Gauge<Long>) () -> garbageCollector.getCollectionTime());
    }
  }
```

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmsServiceMetrics.java (hunk 1 above)
Comment (xieyi888) [reply amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmsServiceMetrics.java:149]: good idea!

Comment (xieyi888) [reply amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmsServiceMetrics.java:149]: > Maybe we should define grabage collector name as a tag?

good idea!

**Comment by xieyi888** [comment]
Thanks @baiyangtx @czy006 
I had fix the comment, please take a look

**Review by czy006 — APPROVED** [review]
LGTM

**Review by tcodehuber — APPROVED** [review]
LGTM

**Comment by zhoujinsong** [comment]
Thanks for the contribution! @xieyi888 
Thanks for the review! @czy006 @tcodehuber @baiyangtx

**Pull request merged**